src/types/Stream.d.ts

```diff
@@ -1,10 +1,27 @@
 import Client from './Client'
+import { ExprArg } from './query'
 
 export interface Subscription {
-  client: Client
+  client: StreamClient
   dispatcher: EventDispatcher
 }
 
 export interface EventDispatcher {
   allowedEvents: string[]
 }
+
+export interface StreamClient {
+  client: Client
+  expression: ExprArg
+  options: Options
+  onEvent: Function
```

Comment:
This is too generic, what should we expect here? what arguments and response?

Reply:
This may help https://cjam.github.io/typesafe-event-emitter-in-typescript/

Reply:
This class is internal. We should focus on the public interface, IMO.